kafka/consumer.go

```diff
@@ -261,6 +263,58 @@ func (c *Consumer) Events() chan Event {
 	return c.events
 }
 
+// ReadMessage polls the consumer for a message.
+//
+// This is a conveniance API that wraps Poll() and only returns
+// messages or errors. All other event types are discarded.
+//
+// The call will block for at most `timeoutMs` waiting for
+// a new message or error. `timeoutMs` may be set to -1 for
+// indefinate wait.
+//
+// Timeout is returned as (nil, err) where err is `kafka.(Error).Code == Kafka.ErrTimedOut`.
+//
+// Messages are returned as (msg, nil),
+// while general errors are returned as (nil, err),
+// and partition-specific errors are returned as (msg, err) where
+// msg.TopicPartition provides partition-specific information (such as topic, partition and offset).
+//
+// All other event types, such as PartitionEOF, AssignedPartitions, etc, are silently discarded.
```

Comment:
this worries me a bit - it'd be preferable to have a single api that can be used for this purpose, but which you can also subscribe to the other event types if required. if that's possible (using channels or something?) i think you don't really want to introduce this.

Reply:
This is a high-level convenience interface.
The thing you are describing is what the current Poll() API combined with a switch statement already provides